src/ripple/protocol/impl/ErrorCodes.cpp

```diff
@@ -136,19 +136,31 @@ sortErrorInfos (ErrorInfo const (&unordered)[N]) -> ErrorInfoArray<N>
 
     // Verify that all entries are filled in starting with 1 and proceeding
     // to rpcLAST.
-    int expect = 0;
+    //
+    // It's okay for there to be missing entries; they will contain the code
+    // rpcUNKNOWN.  But other than that all entries should match their index.
+    int codeCount {0};
+    int expect {rpcBAD_SYNTAX - 1};
```

Comment:
You could add an `rpcFIRST = rpcBAD_SYNTAX` and use that to initialize `expect`, similar to how rpcLAST is used to check bounds above, it could be added there too instead to become `if (info.code < rpcFIRST || info.code > rpcLAST)`

It still has the oddity of the - 1, but I see why it had to be that way.

Reply:
I could.  `rpcBAD_SYNTAX` was _supposed_ to be another magic numeric code (like `rpcUNKNOWN`), so it's value was supposed to be known not to change.  But, as @ximinez recently pointed out to me, that magic is not working at the moment.

Still, I'm not very worried about the value of `rpcBAD_SYNTAX` changing in the future.  I do expect `rpcLAST` to be a moving target, so I think it's important to have that alias to the actual last numeric code.

If you feel strongly about adding `rpcFIRST` I can do that, but my inclination is not to do so.  In my experience alias proliferation can lead to confusion, although I don't think that specific alias would be particularly confusing.

Reply:
Nah, no strong feelings, I just like it when things are symmetrical.